subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java

```diff
@@ -555,7 +553,7 @@ private boolean resolveFromDefaultImports(ClassNode type, boolean testDefaultImp
             String name = type.getName();
             List<String> fqn = simpleNameToFQN.get(type.getName());
             if (fqn != null) {
-                if (fqn.size() == 1 && resolveFromResolver(type, fqn.get(0))) {
```

Comment:
Why is this changed? If we need to change it, we'd better create a test to cover this scenario.